src/plugins/wizard/public/application/contributions/containers/data_tab/items/use/use_dropbox.tsx

```diff
@@ -154,17 +164,36 @@ export const useDropbox = (props: UseDropboxProps): DropboxProps => {
     onDropField
   );
 
-  const isValidDropField = useMemo(() => {
-    if (!dragData) return false;
+  useEffect(() => {
+    const getValidAggTypes = () => {
+      const types: unknown[] = [];
+      if (!dragData || schema.group === 'none') return [];
+
+      const indexField = getIndexPatternField(dragData.name, indexPattern?.fields ?? []);
+
+      if (!indexField) return [];
+
+      // Check if the aggTypes allowed by the schema allow this field
+      const aggTypes = aggService.types.getAll();
+
+      const allowedAggTypes = filterByName(aggTypes[schema.group], schema.aggFilter);
+
+      allowedAggTypes.forEach((aggType) => {
+        const allowedFieldTypes = aggType.paramByName('field')?.filterFieldTypes;
+        const isFieldDroppable = filterByType([indexField], allowedFieldTypes).length !== 0;
```

Comment:
nit - this name doesn't need to include `Droppable` - this test is more generic, something like `isFieldAllowed` or something like that. I also kind of think this would be more readable if written with `reduce()` instead of `forEach()`.

Reply:
Not a fan of reduce, but i do agree with the readability. Ive refactored this to just use the filter and map functions.